File: Pacode/2026.2/turma_intermediaria/ListaIntermediaria_CoordinateCompression_SparseTable/B - Covered Points Count/B-covered-points-count.py
import sys

data = iter(map(int, sys.stdin.buffer.read().split()))
n = next(data)
events = []

for _ in range(n):
    left = next(data)
    right = next(data)
    events.append((left, 1))
    events.append((right + 1, -1))

events.sort()
answer = [0] * (n + 1)
previous = events[0][0]
covering = 0

for x, change in events:
    if covering:
        answer[covering] += x - previous
    covering += change
    previous = x

sys.stdout.write(' '.join(map(str, answer[1:])))

File: Pacode/2026.2/turma_intermediaria/ListaIntermediaria_CoordinateCompression_SparseTable/B - Covered Points Count/B-covered-points-count.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {
    ios::sync_with_stdio(false);
    cin.tie(nullptr);
    int n;
    cin >> n;
    vector<pair<long long, int>> events;
    for (int i = 0; i < n; i++) {
        long long l, r;
        cin >> l >> r;
        events.push_back({l, 1});
        events.push_back({r + 1, -1});
    }
    sort(events.begin(), events.end());
    vector<long long> answer(n + 1);
    long long previous = events[0].first;
    int covering = 0;
    for (int i = 0; i < (int)events.size();) {
        long long x = events[i].first;
        if (covering) answer[covering] += x - previous;
        while (i < (int)events.size() && events[i].first == x) {
            covering += events[i].second;
            i++;
        }
        previous = x;
    }
    for (int i = 1; i <= n; i++) {
        if (i > 1) cout << ' ';
        cout << answer[i];
    }
    cout << '\n';
    return 0;
}
